core/shared/platform/esp-idf/espidf_socket.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2021 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "platform_api_vmcore.h"
+#include "platform_api_extension.h"
+
+#include <arpa/inet.h>
+
+static void
+textual_addr_to_sockaddr(const char *textual, int port, struct sockaddr_in *out)
+{
+    assert(textual);
+
+    out->sin_family = AF_INET;
+    out->sin_port = htons(port);
+    out->sin_addr.s_addr = inet_addr(textual);
+}
+
+int
+os_socket_create(bh_socket_t *sock, int tcp_or_udp)
```

Comment:
Should be `os_socket_create(bh_socket_t *sock, bool is_tcp)`, see definition in platform_api_extension.h